### PR TITLE
doc: Update TF-M documentation to match current TF-M version

### PR DESCRIPTION
**Pull request by joerchan**
Update TF-M documentation to match the current integrated TF-M version.
Include mention of the Platform secure partition that can be enabled.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

File: doc/guides/tfm/overview.rst [hunk 1]
```diff
@@ -196,12 +196,13 @@ within a default Zephyr west workspace.)
 Secure Services
 ---------------
 
-As of TF-M 1.3.0, the following secure services are available:
+As of TF-M 1.4.1, the following secure services are available:
```
Comment (SebastianBoe) [inline doc/guides/tfm/overview.rst:199]: are generally available (although vendor support may vary):

Comment (SebastianBoe) [reply doc/guides/tfm/overview.rst:199]: IAS is not supported for Nordic. Audit and FWU I am unsure of.

Comment (joerchan) [inline doc/guides/tfm/overview.rst:199]:
```suggestion
As of TF-M 1.4.1, the following secure services are generally available (although vendor support may vary):
```
You'd like me to update like this?

Comment (SebastianBoe) [reply doc/guides/tfm/overview.rst:199]: yes

**Review by SebastianBoe — APPROVED** [review]
bumpity bump

**Review by mbolivar-nordic — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Pull request merged**